source/io.c

```diff
@@ -619,39 +619,35 @@ struct aws_napi_input_stream_impl {
     bool eos;          /* end of stream */
 };
 
-static int s_input_stream_seek(struct aws_input_stream *stream, aws_off_t offset, enum aws_stream_seek_basis basis) {
+static int s_input_stream_seek(struct aws_input_stream *stream, int64_t offset, enum aws_stream_seek_basis basis) {
     struct aws_napi_input_stream_impl *impl = stream->impl;
 
     int result = AWS_OP_SUCCESS;
     uint64_t final_offset = 0;
-    int64_t checked_offset = offset;
 
     aws_mutex_lock(&impl->mutex);
     uint64_t total_bytes = impl->bytes_read + impl->buffer.len;
 
-    switch (basis) {
-        case AWS_SSB_BEGIN:
-            /* Offset must be positive, must be greater than the bytes already read (because those
-             * bytes are gone from the buffer), and must not be greater than the sum of the bytes
-             * read so far and the size of the buffer
-             */
-            if (checked_offset < 0 || (uint64_t)checked_offset > total_bytes ||
-                (uint64_t)checked_offset < impl->bytes_read) {
-                result = AWS_IO_STREAM_INVALID_SEEK_POSITION;
-                goto failed;
-            }
-            final_offset = (uint64_t)checked_offset - impl->bytes_read;
-            break;
-        case AWS_SSB_END:
-            /* Offset must be negative, and must not be trying to go further back than the
-             * current length of the buffer, because those bytes have been purged
-             */
-            if (checked_offset > 0 || checked_offset == INT64_MIN || (uint64_t)(-checked_offset) > impl->buffer.len) {
-                result = AWS_IO_STREAM_INVALID_SEEK_POSITION;
-                goto failed;
-            }
-            final_offset = (uint64_t)impl->buffer.len - (uint64_t)(-checked_offset);
-            break;
+    if (basis == AWS_SSB_BEGIN) {
```

Comment:
Should we make this part a public utility function in aws-c-*?

Reply:
So I did that... then I went to incorporate it here, and it turns out that this function is doing something very different. While it clearly started as a copy/paste of that other function, it's doing something different now (can't seek backwards, after a read() it memmoves the remainder of the buffer to the start of the buffer, uses "offsets" to mean "offset from current read position" instead of "offset from start")

Anyway, I went and reverted the utility function I added to aws-c-io. Keeping this as is.

It was a great idea though! I totally agreed until I got to the 95% mark and was like waaaaait a minute.